src/reaktoro_pse/tutorials/softening_acid_ro_example.ipynb

```diff
@@ -0,0 +1,1324 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Reaktoro for modeling softening and acid addition pre-treatment for desalination systems\n",
+    "\n",
+    "\n",
+    "#### Dependencies\n",
+    "* Python - Programming language\n",
+    "* Pyomo - Python package for equation-oriented modeling\n",
+    "* IDAES - Python package extending Pyomo for flowsheet modeling\n",
+    "* cyipopt - Solver necessary for use with gray box models\n",
+    "* Reaktoro-pse - Python package for building Reaktoro gray box models\n",
+    "* watertap-solvers - for cyipopt solver wrapper"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This example demonstrates how Reaktoro blocks can model pre-treatment and other chemistry phenomena in a water treatment train:\n",
+    "\n",
+    "1) Precipitation processes - here modeled as Lime and Soda ash softening for Calcite precipitation. Reaktoro can be configured to use any number of chemicals and users can choose the set of solids to be formed from those available in selected databases.\n",
+    "2) Acidification or pH control - here we add HCl to control effluent pH. Any number of chemicals can be added to modify effluent properties\n",
+    "3) Scaling predictions in desalination, or concentration process - here we monitor formation of Calcite and Gypsum in a hypothetical water concentration process that perfectly removes water from the brine. \n",
+    "\n",
+    "The demo below uses simplified models, but all of these can be replaced with higher fidelity models available in WaterTAP and PrOMMiS"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<img src=\"softening_acid_ro.png\" width=\"1200\" height=\"300\">"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Demonstration structure \n",
+    "* Setting up feed\n",
+    "* Creating simple softening model with Reaktoro\n",
+    "* Creating simple acid addition model with Reaktoro\n",
+    "* Adding RO model, with scaling calculation using Reaktoro"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Example for speciation of NaCl and use with RO model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Import needed modules"
```

Comment:
```suggestion
    "### Step 0: Import needed modules"
```
Number the MAJOR steps outlined above in "Demonstration Structure" would be helpful. I would also suggest condensing some of these separate cells by aggregating where viable.